core/src/main/java/com/klaviyo/coresdk/ConfigFileUtils.kt

```diff
@@ -0,0 +1,45 @@
+package com.klaviyo.coresdk
+
+import java.io.File
+import java.util.UUID
+
+internal object ConfigFileUtils {
+    private const val CONFIG_FILE_NAME = "klaviyo-sdk-config"
+
+    private const val UUID_KEY = "uuid"
+
+    fun readOrCreateUUID(): String {
+        val file = File("${KlaviyoConfig.applicationContext.filesDir}/$CONFIG_FILE_NAME")
+
+        var uuid = findConfigValue(file, UUID_KEY)
+
+        if (uuid.isNullOrEmpty()) {
+            uuid = UUID.randomUUID().toString()
+            writeConfigValue(file, UUID_KEY, uuid)
+        }
+
+        return uuid
+    }
+
+    fun findConfigValue(file: File, key: String): String? {
+        var value: String? = null
+
+        if (!file.exists()) {
+            return null
+        }
+
+        file.useLines { lines ->
+            lines.forEach { line ->
+                if (line.startsWith("$key:")) {
+                    value = line.substring("$key:".length)
+
+                }
+            }
+        }
+        return value
+    }
+
+    fun writeConfigValue(file: File, key:String, value: String) {
```

Comment:
should this be private?

Reply:
The encompassing object is internal so there's no worry about it showing up as a function in the API. Left the function calls like this for any future file writing that the SDK needs. Acting as a utility class here

Reply:
Could totally privatize too though. Just makes testing harder